src/catkin_pkg/packages.py

```diff
@@ -104,11 +115,12 @@ def find_packages_allowing_duplicates(basepath, exclude_paths=None, exclude_subs
     :param warnings: Print warnings if None or return them in the given list
     :returns: A dict mapping relative paths to ``Package`` objects ``dict``
     """
-    packages = {}
     package_paths = find_package_paths(basepath, exclude_paths=exclude_paths, exclude_subspaces=exclude_subspaces)
-    for path in package_paths:
-        packages[path] = parse_package(os.path.join(basepath, path), warnings=warnings)
-    return packages
+    parser = _PackageParser(capture_warnings=isinstance(warnings, list))
+    results = multiprocessing.Pool(4).map(parser, package_paths)
+    if parser.capture_warnings:
+        map(warnings.extend, [package_warnings for _, _, package_warnings in results])
```

Comment:
Okay, I've dropped the comprehensions in favour of zipping; I think that's maybe a bit clearer.